core/src/main/scala/codecs/codecs.scala

```diff
@@ -51,72 +52,76 @@ package object codecs extends lowerprioritycodecs with TupleHelpers {
   implicit val utf8 = C.variableSizeBytes(int32, C.utf8)
   implicit val bool = C.bool(8) // use a full byte
 
-  implicit def tuple2[A:Codec,B: Codec]: Codec[(A,B)] =
-    Codec[A] ~ Codec[B]
+  implicit def tuple2[A, B](implicit LCA: Lazy[Codec[A]], LCB: Lazy[Codec[B]]): Lazy[Codec[(A,B)]] =
```

Comment:
@jedesah I tested it out, seems like to be a shapeless issue that most likely is solved in 2.2. So disregard my comment about changing to 1.7.1.